gnomad_constraint/utils/constraint.py

```diff
@@ -391,6 +398,208 @@ def apply_models(
+        Table.
+    :param ac_cutoff: The cutoff of allele count when filtering context Table and genome sites Table.
+    :param downsampling_level: The size of downsamplings will be used to count variants. Default is 1000.
+    :param total_mu: The per-generation mutation rate. Default is 1.2e-08.
+    :param pops: List of populations to use for downsampling counts. Default is ().
+    :return: Mutation rate Table.
+    """
+    # keep only loci where the mean coverage in the gnomAD genomes was between 15X and
+    # 60X.
+    context_ht = filter_to_autosomes(remove_coverage_outliers(raw_context_ht))
+    genome_ht = filter_to_autosomes(remove_coverage_outliers(genome_ht))
+
+    # filter the Table so that the most severe annotation was intron_variant or
+    # intergenic_variant, and that the GERP score was between the 5th and 95th
+    # percentile of the genomewide distribution.
+    context_ht = filter_for_mu(context_ht)
+    genome_ht = filter_for_mu(genome_ht)
+
+    context_ht = context_ht.select(*keep_annotations)
+    genome_ht = genome_ht.select(*list(keep_annotations) + ["freq", "pass_filters"])
+
+    # downsampled the dataset to 1,000 genomes
+    freq_index = hl.eval(
+        hl.enumerate(genome_ht.freq_meta).find(
+            lambda f: (f[1].get("downsampling") == str(downsampling_level))
+            & (f[1].get("pop") == "global")
+            & (f[1].get("group") == "adj")
+            & (f[1].size() == 3)
+        )
+    )[0]
+    freq_expr = genome_ht.freq[freq_index]
+    # Set up the criteria to filtered out  low quality sites, and sites found in
+    # greater than 5 copies in the downsampled set.
+    keep_criteria = (freq_expr.AC <= ac_cutoff) & genome_ht.pass_filters
+
+    # Only keep variants with AC <= ac_cutoff and passing filters in genome site
+    # Table.
+    genome_ht = genome_ht.filter(keep_criteria)
+
+    # In the denominator, only keep variants not in the genome dataset, or with AC
+    # <= ac_cutoff and passing filters.
+    context_ht = context_ht.anti_join(genome_ht.filter(keep_criteria, keep=False))
+
+    if not summary_file:
+        summary_file = new_temp_file(prefix="constraint", extension="he")
+
+    all_possible_dtype = count_variants_by_group(
+        context_ht,
+        omit_methylation=omit_methylation,
+        return_type_only=True,
+    ).variant_count.dtype
+
+    # Count possible variants in context Table.
+    if recalculate_all_possible_summary:
+        all_possible = count_variants_by_group(
+            context_ht,
+            omit_methylation=omit_methylation,
+        ).variant_count
+        # with hl.hadoop_open(summary_file, "wb") as f:
+        #     pickle.dump(all_possible, f)
+        hl.experimental.write_expression(all_possible, summary_file)
+    # with hl.hadoop_open(summary_file, "rb") as f:
+    #     all_possible = pickle.load(f)
+    all_possible = hl.experimental.read_expression(summary_file)
+
```

Comment:
Add in a resource function like this:

def get_checkpoint_path(
    name: str, version: str = CURRENT_VERSION, extension: str = "ht"
) -> str:
    """
    Create a checkpoint path for Table or MatrixTable.
    """
    return f'{constraint_tmp_prefix}/{version}/checkpoint_files/{name}.{extension}'

Reply:
added this to the resource file although changed extension to mt to make similar to what we have in gnomac_qc